charmcraft/charm_builder.py

```diff
@@ -365,13 +365,13 @@ def _parse_arguments() -> argparse.Namespace:
         help="The charm entry point. Default is 'src/charm.py'.",
     )
     parser.add_argument(
-        "--charmdir",
+        "--builddir",
```

Comment:
How much will this affect the end user?

I wonder if it is worth having a deprecation notice or notifying users to use the new arguments.  The `--builddir` parameter keeping the same name but changing its meaning could be confusing.

Reply:
Discussed offline - `charm_builder` isn't called by the user.

Reply:
no final user involved! :) this is deep in the execution layers of `pack` command